Not able to select all tables for Oracle Database

**Describe the bug**
When trying to run integration tests with Oracle Database, the following exception occurs:

```
org.operaton.bpm.engine.ProcessEngineException: ENGINE-03019 Unable to fetch process engine table names.

	at org.operaton.bpm.engine.impl.db.EnginePersistenceLogger.getDatabaseTableNameException(EnginePersistenceLogger.java:247)
	at org.operaton.bpm.engine.impl.db.sql.DbSqlSession.getTableNamesPresent(DbSqlSession.java:648)
	at org.operaton.bpm.engine.impl.db.entitymanager.DbEntityManager.getTableNamesPresentInDatabase(DbEntityManager.java:749)
	at org.operaton.bpm.engine.impl.cmd.PurgeDatabaseAndCacheCmd.purgeDatabase(PurgeDatabaseAndCacheCmd.java:79)
	at org.operaton.bpm.engine.impl.cmd.PurgeDatabaseAndCacheCmd.execute(PurgeDatabaseAndCacheCmd.java:62)
	at org.operaton.bpm.engine.impl.cmd.PurgeDatabaseAndCacheCmd.execute(PurgeDatabaseAndCacheCmd.java:45)
	at org.operaton.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:29)
	at org.operaton.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:113)
	at org.operaton.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:64)
	at org.operaton.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:34)
	at org.operaton.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:34)
	at org.operaton.bpm.engine.impl.interceptor.ExceptionCodeInterceptor.execute(ExceptionCodeInterceptor.java:55)
	at org.operaton.bpm.engine.impl.ManagementServiceImpl.purge(ManagementServiceImpl.java:294)
	at org.operaton.bpm.engine.impl.test.TestHelper.assertAndEnsureCleanDbAndCache(TestHelper.java:359)
	at org.operaton.bpm.engine.impl.test.TestHelper.assertAndEnsureCleanDbAndCache(TestHelper.java:340)
	at org.operaton.bpm.engine.test.junit5.ProcessEngineExtension.afterTestExecution(ProcessEngineExtension.java:261)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.sql.SQLException: ORA-17041: Fehlender IN- oder OUT-Parameter auf Index: 1
https://docs.oracle.com/error-help/db/ora-17041/
	at oracle.jdbc.driver.OraclePreparedStatement.processCompletedBindRow(OraclePreparedStatement.java:2067)
	at oracle.jdbc.driver.OraclePreparedStatement.executeInternal(OraclePreparedStatement.java:3940)
	at oracle.jdbc.driver.OraclePreparedStatement.executeQuery(OraclePreparedStatement.java:4142)
	at oracle.jdbc.driver.OraclePreparedStatementWrapper.executeQuery(OraclePreparedStatementWrapper.java:1103)
	at org.operaton.bpm.engine.impl.db.sql.DbSqlSession.getTablesPresentInOracleDatabase(DbSqlSession.java:661)
	at org.operaton.bpm.engine.impl.db.sql.DbSqlSession.getTableNamesPresent(DbSqlSession.java:614)
	... 16 more

```

**To Reproduce**
Check out branch `feature/enable-oracle-testcontainers` and run `mvn clean install -Pdistro,oracle` or start the engine with a connection to Oracle DB


The issue was probably introduced in a refactoring with PR #493. After the refactoring the statement is executed and the parameter is set afterwards, leading to the aforementioned error

--- a/database/pom.xml
+++ b/database/pom.xml
@@ -309,6 +309,8 @@
         <database.type>oracle</database.type>
         <database.driver>oracle.jdbc.OracleDriver</database.driver>
         <database.datasource.class>oracle.jdbc.pool.OracleDataSource</database.datasource.class>
+        <database.tc.url>operatonoracle:21-faststart</database.tc.url>
+        <database.url>jdbc:tc:${database.tc.url}://localhost:${database.port}${database.tc.params}</database.url>
       </properties>
       <dependencies>
         <dependency>

--- a/engine/src/main/java/org/operaton/bpm/engine/impl/db/sql/DbSqlSession.java
+++ b/engine/src/main/java/org/operaton/bpm/engine/impl/db/sql/DbSqlSession.java
@@ -657,17 +657,18 @@ public abstract class DbSqlSession extends AbstractPersistenceSession {
     String databaseTablePrefix = getDbSqlSessionFactory().getDatabaseTablePrefix();
 
     try(Connection connection = Context.getProcessEngineConfiguration().getDataSource().getConnection();
-        PreparedStatement prepStat = connection.prepareStatement(selectTableNamesFromOracle);
-        ResultSet tablesRs = prepStat.executeQuery()) {
+        PreparedStatement prepStat = connection.prepareStatement(selectTableNamesFromOracle)) {
 
       prepStat.setString(1, databaseTablePrefix + "ACT-_%");
 
-      while (tablesRs.next()) {
-        String tableName = tablesRs.getString("TABLE_NAME");
-        tableName = tableName.toUpperCase();
-        tableNames.add(tableName);
+      try(ResultSet tablesRs = prepStat.executeQuery()) {
+        while (tablesRs.next()) {
+          String tableName = tablesRs.getString("TABLE_NAME");
+          tableName = tableName.toUpperCase();
+          tableNames.add(tableName);
+        }
+        LOG.fetchDatabaseTables("oracle all_tables", tableNames);
       }
-      LOG.fetchDatabaseTables("oracle all_tables", tableNames);
     }
 
     return tableNames;

--- a/test-utils/testcontainers/pom.xml
+++ b/test-utils/testcontainers/pom.xml
@@ -54,6 +54,10 @@
       <groupId>org.testcontainers</groupId>
       <artifactId>mssqlserver</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>oracle-xe</artifactId>
+    </dependency>
 
     <!-- test dependencies -->
     <dependency>

--- a/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
+++ b/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
@@ -32,7 +32,7 @@ public class OperatonOracleContainerProvider extends OracleContainerProvider {
 
   @Override
   public JdbcDatabaseContainer<?> newInstance(String tag) {
-    if("aarch64".equals(System.getProperty("os.arch"))) {
+    if("aaarch64".equals(System.getProperty("os.arch"))) {
       throw new IllegalStateException("The Oracle Testcontainers tests cannot be executed on ARM architecture, as Oracle Database does not support it.");
     }
     DockerImageName dockerImageName = TestcontainersHelper

--- a/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
+++ b/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
@@ -1,0 +1,42 @@
+/*
+ * Copyright 2025 the Operaton contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.operaton.impl.test.utils.testcontainers;
+
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.OracleContainer;
+import org.testcontainers.containers.OracleContainerProvider;
+import org.testcontainers.utility.DockerImageName;
+
+public class OperatonOracleContainerProvider extends OracleContainerProvider {
+
+  private static final String NAME = "operatonoracle";
+
+  @Override
+  public boolean supports(String databaseType) {
+    return NAME.equals(databaseType);
+  }
+
+  @Override
+  public JdbcDatabaseContainer<?> newInstance(String tag) {
+    if("aarch64".equals(System.getProperty("os.arch"))) {
+      throw new IllegalStateException("The Oracle Testcontainers tests cannot be executed on ARM architecture, as Oracle Database does not support it.");
+    }
+    DockerImageName dockerImageName = TestcontainersHelper
+      .resolveDockerImageName("oracle", tag, "gvenzl/oracle-xe");
+    return new OracleContainer(dockerImageName);
+  }
+}

--- a/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
+++ b/test-utils/testcontainers/src/main/java/org/operaton/impl/test/utils/testcontainers/OperatonOracleContainerProvider.java
@@ -32,9 +32,6 @@ public class OperatonOracleContainerProvider extends OracleContainerProvider {
 
   @Override
   public JdbcDatabaseContainer<?> newInstance(String tag) {
-    if("aaarch64".equals(System.getProperty("os.arch"))) {
-      throw new IllegalStateException("The Oracle Testcontainers tests cannot be executed on ARM architecture, as Oracle Database does not support it.");
-    }
     DockerImageName dockerImageName = TestcontainersHelper
       .resolveDockerImageName("oracle", tag, "gvenzl/oracle-xe");
     return new OracleContainer(dockerImageName);

--- a/test-utils/testcontainers/src/main/resources/META-INF/services/org.testcontainers.containers.JdbcDatabaseContainerProvider
+++ b/test-utils/testcontainers/src/main/resources/META-INF/services/org.testcontainers.containers.JdbcDatabaseContainerProvider
@@ -2,3 +2,4 @@ org.operaton.impl.test.utils.testcontainers.OperatonPostgreSQLContainerProvider
 org.operaton.impl.test.utils.testcontainers.OperatonMariaDBContainerProvider
 org.operaton.impl.test.utils.testcontainers.OperatonMySqlContainerProvider
 org.operaton.impl.test.utils.testcontainers.OperatonMSSQLContainerProvider
+org.operaton.impl.test.utils.testcontainers.OperatonOracleContainerProvider

